fairlearn/metrics/_metric_frame.py

```diff
@@ -323,14 +330,105 @@ def __init__(
                 raise ValueError(_DUPLICATE_FEATURE_NAME.format(name))
             nameset.add(name)
 
-        # Create the 'overall' results
-        self._result = DisaggregatedResult.create(
+        # Create the basic results
+        result = DisaggregatedResult.create(
             data=all_data,
             annotated_functions=annotated_funcs,
             sensitive_feature_names=self._sf_names,
             control_feature_names=self._cf_names,
         )
 
+        # Build into cache
+        self._result_cache = dict()
+        self._populate_results(result)
+
+    def _extract_result(self, underlying_result, no_control_levels: bool):
```

Comment:
I suppose this is here also to keep backward compatibility?

Reply:
You could say that. Most of the work has been shuffled off into the `DisaggregatedResult`, and that tends towards my preference for type safety. The functionality here isn't new, but was previously distributed, to supply the more 'user friendly' return values of `MetricFrame`